publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/TrustedThirdPartyStrategyTest.java

```diff
@@ -0,0 +1,119 @@
+
+    @ParameterizedTest(name = "Should deny trusted third party {0} operation on non-degree resources")
+    @EnumSource(value = PublicationOperation.class, mode = Mode.EXCLUDE, names = {"UNPUBLISH", "UPDATE", "TERMINATE",
+        "TICKET_PUBLISH"})
+    void shouldDenyTrustedThirdPartyOnNonDegree(PublicationOperation operation)
+        throws JsonProcessingException, UnauthorizedException {
+
+        var requestInfo = createThirdPartyRequestInfo();
+        var publication = createNonDegreePublication(randomString(), EXTERNAL_CLIENT_CUSTOMER_URI);
+        var userInstance = RequestUtil.createUserInstanceFromRequest(requestInfo, identityServiceClient);
+
+        Assertions.assertFalse(PublicationPermissionStrategy
+                                   .create(publication, userInstance, uriRetriever)
+                                   .allowsAction(operation));
+    }
+
+    @Test
+    void shouldAllowTrustedThirdPartyDeleteOnDraftNonDegree()
+        throws JsonProcessingException, UnauthorizedException {
+
+        var requestInfo = createThirdPartyRequestInfo();
+        var userInstance = RequestUtil.createUserInstanceFromRequest(requestInfo, identityServiceClient);
+        var publication =
+            createNonDegreePublication(randomString(), EXTERNAL_CLIENT_CUSTOMER_URI,
+                                       userInstance.getTopLevelOrgCristinId()).copy()
+                .withStatus(DRAFT)
+                .build();
+
+        Assertions.assertTrue(PublicationPermissionStrategy
+                                  .create(publication, userInstance, uriRetriever)
+                                  .allowsAction(DELETE));
+    }
+
+    @Test
+    void shouldDenyTrustedThirdPartyWithMissingTopLevelOrgCristinId() {
+
+        var userInstance = mock(UserInstance.class);
+        when(userInstance.isExternalClient()).thenReturn(true);
+
+        var publication =
+            createNonDegreePublication(randomString(), EXTERNAL_CLIENT_CUSTOMER_URI,
+                                       randomUri()).copy()
+                .withStatus(UNPUBLISHED)
+                .build();
+
+        Assertions.assertFalse(PublicationPermissionStrategy
+                                  .create(publication, userInstance, uriRetriever)
+                                  .allowsAction(DELETE));
+    }
+
+    @Test
+    void shouldDenyTrustedClientEditPublicationWithMissingPublisher()
+        throws JsonProcessingException, UnauthorizedException {
+        var publication = createNonDegreePublication(randomString(), randomUri());
+        publication.setPublisher(null);
+        var requestInfo = createThirdPartyRequestInfo();
+
+        Assertions.assertFalse(
+            PublicationPermissionStrategy.create(publication, RequestUtil.createUserInstanceFromRequest(
+                    requestInfo, identityServiceClient), uriRetriever)
+                .allowsAction(UPDATE));
+    }
+
+    @Test
+    void shouldAllowTrustedClientEditPublicationOnDegree()
```

Comment:
On _any_ degree?

Reply:
as long as userInstance.getCustomerId().equals(publication.getPublisher().getId()), as defined in TrustedThirdPartyStrategy. this is not the best check, but probably close enough

Reply:
will update title of test

Reply:
Ah, must admit I didn't see that bit. My bad. That looks sound! Would a test `shouldDenyTrustedClientEditPublicationOnDegreeFromAnotherPublisher()` be a good addition? I can see that you have a test for if `publisher == null`, which might be enough? 🤷‍♂️ Your call.

Reply:
added shouldDenyTrustedClientEditPublicationOnDegreeFromAnotherPublisher 👍